packages/fuels-core/src/code_gen/abigen.rs

```diff
@@ -172,11 +174,17 @@ impl Abigen {
                 pub struct #name{
                     wallet: WalletUnlocked,
                     binary_filepath: String,
+                    logs_map: HashMap<(Bech32ContractId, u64), ParamType>,
                 }
 
                 impl #name {
                     pub fn new(wallet: WalletUnlocked, binary_filepath: &str) -> Self {
-                        Self {wallet: wallet, binary_filepath: binary_filepath.to_string()}
+                        Self {
+                            wallet: wallet,
+                            binary_filepath: binary_filepath.to_string(),
+                            logs_map: get_logs_hashmap(&[#(#log_id_param_type_pairs),*],
+                                                       &Bech32ContractId::from(ContractId::zeroed()))
```

Comment:
This makes me think we should accept an `Option<Bech32ContractID>` for the `get_logs_hashmap` function: it makes no sense to pass in a null contract_id, it should be hidden away in the function itself. Wdyt?
Also this way we don't have to have the `Bech32ContractID` include statement in the general case, which makes more sense.